libbeat/kibana/index_pattern_generator_test.go

```diff
@@ -227,18 +226,25 @@ func find(a []map[string]interface{}, key, val string) int {
 	return -1
 }
 
-func readJson(path string) (map[string]interface{}, error) {
+func readNdJson(path string) ([]map[string]interface{}, error) {
 	f, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
 
-	var data map[string]interface{}
-	err = json.Unmarshal(f, &data)
-	if err != nil {
-		return nil, err
+	data := make([]map[string]interface{}, 0)
+	d := json.NewDecoder(bytes.NewReader(f))
+	for {
```

Comment:
I've used the `More()` method for this sort of streaming json parsing in the past. Then I think you can skip the io.EOF check.

	dec := json.NewDecoder(bytes.NewReader(f))
	for dec.More() {
                var foo Foo
		if err := dec.Decode(&foo); err != nil {
			log.Fatal(err)
		}
	}